### PR TITLE
feat(governance): referral proposals

### LINKED ISSUE #4397: Show referral programme proposals on governance app
## Story

As a user
I want to see when a proposal has been made for a referral programme 
So that I can review and vote

## Acceptance Criteria 

- [ ] The governance app is able to handle new governance proposal types for referral program 
- [ ] On the home page and proposal listing page summary, I can see which proposals are for the referral programme 
- [ ] I can see the usual voting information
- [ ] I can see the usual description info
- [ ] I can see the "Program Details", ie. the benefit tiers in some sort of nice UI components (shown in screenshot below)
- [ ] I can see full proposal JSON at the bottom as usual

## Tasks

- [x] Specs reviewed and created or adjusted
- [ ] Implementation
- [ ] Testing (unit and/or e2e)
- [ ] Code review
- [ ] QA review

## Sketch

This is information that will be available per tier in a new API to be built by Core.

Image

This is how it is laid out in the referral scheme page

Image

We probably need some sort of simplified version of this table(s) to display this in the governance app.

## Additional details / background info

Spec: https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0083-RFPR-on_chain_referral_program.md

Core API Ticket: https://github.com/vegaprotocol/vega/issues/9255

### PR DESCRIPTION
**Pull request by sam-keen**
# Related issues 🔗

Closes #4397

Provides information on referral proposals. I've added tooltip descriptions for most of the table keys based on the data in the specs

Screenshot 2023-10-10 at 16 51 11

Screenshot 2023-10-10 at 16 51 22

### CODE REVIEW

**Review by jeremyletang — APPROVED** [review]

File: apps/governance/src/routes/proposals/proposals/Proposals.graphql [hunk 1]
```diff
@@ -43,6 +43,31 @@ fragment UpdateMarketStates on Proposal {
   }
 }
 
+fragment UpdateReferralPrograms on Proposal {
+  terms {
+    change {
+      ... on UpdateReferralProgram {
+        changes {
+          id
+          version
```
Comment (ValentinTrinque) [inline apps/governance/src/routes/proposals/proposals/Proposals.graphql:52]: `id` and `version` are internal values, and shouldn't be needed. It's a bug from the gql schema.

File: apps/governance/src/routes/proposals/components/proposal-referral-program-details/proposal-referral-program-details.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+import BigNumber from 'bignumber.js';
+import { useAppState } from '../../../../contexts/app-state/app-state-context';
+
+interface ProposalReferralProgramDetailsProps {
+  proposal: ProposalQuery['proposal'];
+}
+
+export const formatEndOfProgramTimestamp = (value: string) => {
+  return formatDateWithLocalTimezone(new Date(value));
+};
```
Comment (edd) [inline apps/governance/src/routes/proposals/components/proposal-referral-program-details/proposal-referral-program-details.tsx:24]: **nitpick (non-blocking)**:  These could be more defensive. I realise Typescript *feels* like it means that these values can never be `null` or `undefined` but changes between versions or just plain old data node bugs can mean that they end up being unexpected values that will throw errors. `try`/`catch`ing or  testing that a string is valid would be good... but at least some of the underlying functions probably already do, so I've marked it as non-blocking.

**Review by edd — APPROVED** [review]
Approved, with one comment about defensiveness in some functions. Generally it's pretty defensive though.

File: apps/governance/src/routes/proposals/proposals/Proposals.graphql (hunk 1 above)
Comment (sam-keen) [reply apps/governance/src/routes/proposals/proposals/Proposals.graphql:52]: ok thanks @ValentinTrinque

**Pull request merged**